src/libs/actions/IOU.js

```diff
@@ -2267,7 +2267,7 @@ function deleteMoneyRequest(transactionID, reportAction, isSingleTransactionView
     // STEP 2: Decide if we need to:
     // 1. Delete the transactionThread - delete if there are no visible comments in the thread
     // 2. Update the moneyRequestPreview to show [Deleted request] - update if the transactionThread exists AND it isn't being deleted
-    const shouldDeleteTransactionThread = transactionThreadID ? ReportActionsUtils.getLastVisibleMessage(transactionThreadID).lastMessageText.length === 0 : false;
+    const shouldDeleteTransactionThread = transactionThreadID ? !reportAction.childVisibleActionCount : false;
```

Comment:
I think we only need this change to get the expected result, no?

Reply:
>So now if we send a message and some modified message and then we delete the request, the modified messages are still visible, right?

On the latest behavior, the report is deleted so the message won't be shown.

Reply:
@mollfpr I am sorry, but this change is match my [proposal](https://github.com/Expensify/App/issues/25997#issuecomment-1704062212) and please look at my last line of my [comment](https://github.com/Expensify/App/issues/25997#issuecomment-1709828282) .

Reply:
@dukenv0307 Could you explain more about what the behaviors that changed?
Could you attach video comparison for the latest behavior and before the change?
Thank you

Reply:
@tsa321 The original issue is removing the modified expense message while retaining the thread report. But now the BE is deleting the thread report if there's no message or only a modified expense message.

@jasperhuangg I'm unsure what action should be taken if the current changes are the same as other proposals. What do you think?

Reply:
@tsa321 I understand the change now matches your proposal, but your proposal didn't fix the original issue that we were trying to fix.

We've already accepted @dukenv0307 to work on this issue, and we therefore aren't considering other contributors, even if the issue changes in such a way that leads to us using a solution that matches your proposal.

Please refrain from commenting on other contributor PRs like this after they've already been accepted to work on an issue–it isn't productive and distracts from collaboration.

Reply:
Oh Ok, I understand...

Reply:
@dukenv0307 I'm asking for clarification on how best to proceed.